### PR TITLE
Chainparams: Wallet: Decouple DefaultFallbackfee() from IsTestChain()

### PR DESCRIPTION
**Pull request by jtimon**
This doesn't change functionality beyond improving the --help tdocumentation.

Together with #16770 , this allows to get rid of Params().IsTestChain().

This is an alternative to #16524 , which, for example @instagibbs doesn't seem to like very much.

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -4420,20 +4420,18 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
...
     }
+    if (nFeePerK > HIGH_TX_FEE_PER_KB) {
+        chain.initWarning(AmountHighWarn("-fallbackfee").translated + " " +
+                          _("This is the transaction fee you may pay when fee estimates are not available.").translated);
+    }
+    walletInstance->m_fallback_fee = CFeeRate(nFeePerK);
+    walletInstance->m_allow_fallback_fee = nFeePerK != 0; //disable fallback fee in case value was set to 0, enable if non-null value
+
     if (gArgs.IsArgSet("-discardfee")) {
         CAmount nFeePerK = 0;
```
Comment (practicalswift) [inline src/wallet/wallet.cpp]: This `nFeePerK` is shadowing the `nFeePerK` in the outer scope.

Comment (jtimon) [reply src/wallet/wallet.cpp]: Yes, it is. And more of the same below. But in every use of their own nFeePerK they initialize it.
Anyway, what do you think I should do?
Perhaps just rename the variable? any suggestion on the name?

Comment (practicalswift) [reply src/wallet/wallet.cpp]: I would use another variable name to disambiguate :)

Comment (jtimon) [reply src/wallet/wallet.cpp]: Fixed nit, renamed to fee_per_k.

**Comment by jtimon** [comment]
Closing since competing https://github.com/bitcoin/bitcoin/pull/16524 was merged instead.

**Pull request closed without merging**